src/main/kotlin/com/team4099/robot2023/subsystems/vision/camera/CameraIONorthstar.kt

```diff
@@ -0,0 +1,84 @@
+package com.team4099.robot2023.subsystems.vision.camera
+
+import com.team4099.robot2023.config.constants.FieldConstants
+import com.team4099.robot2023.util.Alert
+import edu.wpi.first.networktables.DoubleArraySubscriber
+import edu.wpi.first.networktables.IntegerSubscriber
+import edu.wpi.first.networktables.NetworkTableInstance
+import edu.wpi.first.networktables.PubSubOption
+import edu.wpi.first.wpilibj.Timer
+import org.team4099.lib.units.base.Time
+import org.team4099.lib.units.base.inMeters
+import org.team4099.lib.units.base.seconds
+import org.team4099.lib.units.milli
+
+class CameraIONorthstar(val identifier: String) : CameraIO {
+  private var cameraId = 0
+  private var cameraResolutionWidth = 1600
+  private var cameraResolutionHeight = 1200
+  private var cameraAutoExposure = 1
+  private var cameraExposure = 10
+  private var cameraGain = 25
+
+  private val observationSubscriber: DoubleArraySubscriber
+  private val fpsSubscriber: IntegerSubscriber
+
+  private val disconnectedTimeout = 0.5
+  private val disconnectedAlert: Alert
+  private val disconnectedTimer = Timer()
+
+  init {
+    val northStarTable = NetworkTableInstance.getDefault().getTable(identifier)
```

Comment:
Does all of this work on construction or do we need to wait for `robotInit` for network tables to work?